src/util/error.cpp

```diff
@@ -8,14 +8,53 @@ extern "C" {
 TError::TError() : Error(EError::Success) {
 }
 
-TError::TError(EError e, std::string description, int eno) :
+TError::TError(EError e, const std::string &description, int eno) :
     Error(e), Description(description), Errno(eno) {
 }
 
-TError::TError(EError e, int eno, std::string description) :
+TError::TError(EError e, int eno, const std::string &description) :
```

Comment:
TError objects used to return results from many functions, and every object holds a string, even if it describes a errno.
How do you thinks, can we avoid storing description in every object?